subsys/net/lib/lwm2m/lwm2m_registry.c

```diff
@@ -589,6 +601,33 @@ static int lwm2m_engine_set(const char *pathstr, const void *value, uint16_t len
 		*((uint8_t *)data_ptr) = *(uint8_t *)value;
 		break;
 
+	case LWM2M_RES_TYPE_TIME:
+		if (!lwm2m_validate_time_resource_lenghts(max_data_len, len)) {
+			LOG_ERR("Time Set: buffer length %u  max data len %u not supported", len,
+				max_data_len);
+			return -EINVAL;
+		}
+
+		if (max_data_len == sizeof(time_t)) {
+			if (len == sizeof(time_t)) {
+				*((time_t *)data_ptr) = *(time_t *)value;
+			} else {
+				LOG_WRN("Time Set by 32-bit to 64-bit resource:%s", pathstr);
```

Comment:
I'm not sure if this should be a warning - we don't lose any data here, right?

Reply:
True. I will remove that.